eng/common/TestResources/New-TestResources.ps1

```diff
@@ -385,7 +385,15 @@ foreach ($templateFile in $templateFiles) {
 
     Log "Deploying template '$templateFile' to resource group '$($resourceGroup.ResourceGroupName)'"
     $deployment = Retry {
-        New-AzResourceGroupDeployment -Name $BaseName -ResourceGroupName $resourceGroup.ResourceGroupName -TemplateFile $templateFile -TemplateParameterObject $templateFileParameters
+        $lastDebugPreference = $DebugPreference
+        try {
```

Comment:
Why can't you simply pass `-Debug` do this cmdlet? Doesn't that equate to the same thing without the need for the try/finally block?

Reply:
In my testing, adding `-Debug` sets $DebugPreference to "Inquire" for the rest of the script context. This does two undesired things: we don't want debug output for the rest of the script, and the Inquire setting requires manual confirmation, as opposed to "Continue".

Reply:
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_commonparameters?view=powershell-7.1#debug

That seems to suggest that it will set it to Continue in non-interactive mode. I'm not even sure what Inquire does. 

It seems weird that it isn't scoped to just the one cmdlet call but if that is the case what you have works.